pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -103,7 +105,15 @@ func (l *LocalPipelineRefResolver) GetPipeline(ctx context.Context, name string)
 	if l.Namespace == "" {
 		return nil, fmt.Errorf("Must specify namespace to resolve reference to pipeline %s", name)
 	}
-	return l.Tektonclient.TektonV1beta1().Pipelines(l.Namespace).Get(ctx, name, metav1.GetOptions{})
+
+	pipeline, err := l.Tektonclient.TektonV1beta1().Pipelines(l.Namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	if err := verifyResolvedPipeline(ctx, pipeline.DeepCopy()); err != nil {
```

Comment:
It's a bit odd that we are calling verifyResolvedPipeline in two places. Is there any downside to centralizing the verification in one place here: https://github.com/dibyom/pipeline/blob/00b9e9d47a0e788fd88f21fa3527734d8f3a1d8f/pkg/reconciler/pipelinerun/pipelinespec/pipelinespec.go#L63

Reply:
Yeah that's a great suggestion! Previously there are setdefaults() called there, so to not break the current behaviour I didn't move those call into a central place. But now it should be ok to do so with Billy's PR

Reply:
But I think maybe there's one reason I need to to the verification here:
https://github.com/tektoncd/pipeline/blob/b648a5b2eaebf0ece7199dcf2bd89c65137973d5/pkg/reconciler/taskrun/resources/taskref.go#L139-L151

Chuang's working on a PR to get ConfigSource from the `resolver.Get`, it will contain source information which will be used for the next PR's work. In the VerificationPolicy we can use the source url to filter out resources.

Reply:
If the verification happen later then it may not be so easy to get this information?

Reply:
added a todo to address